### PR TITLE
feat: migrate token creation panel to mantine modal

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #5385 
Closes: #5316

### Description:

New Token modal: 
* token creation successful
* token creation error

https://github.com/lightdash/lightdash/assets/7611706/c0417c77-1f09-4ed3-bc04-f250833125ad

### CODE REVIEW

**Comment by joaoviana** [comment]
override `Select` label's marginBottom: 
image

(cc @IrakliJani  & @yasmin-e )

**Comment by IrakliJani** [comment]
Total imports from all libraries: 745 (75.0%)

¼ gone!

File: packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+            label: '90 days',
+            value: '90',
+        },
+    ];
+
+    useEffect(() => {
+        if (isSuccess) {
+            setExpireDate(data.expiresAt);
+        }
+    }, [data?.expiresAt, isSuccess]);
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx:69]: can you move this to the mutation's onSuccess callback? you can make `useCreateAccessToken` accept mutation args. there's a useAccessToken above that function which accepts query options. in this case it should accept mutation options

File: packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    const handleOnSubmit = form.onSubmit(({ description, expiresAt }) => {
+        const currentDate = new Date();
+        const dateWhenExpires = !!Number(expiresAt)
+            ? new Date(
+                  currentDate.setDate(
+                      currentDate.getDate() + Number(expiresAt),
+                  ),
+              )
+            : undefined;
+        setExpireDate(dateWhenExpires);
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx:80]: do we need this state? especially on submit and I see it clears up after successful query

File: packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                                        label={copied ? 'Copied' : 'Copy'}
+                                        withArrow
+                                        position="right"
+                                    >
+                                        <ActionIcon
+                                            color={copied ? 'teal' : 'gray'}
+                                            onClick={copy}
+                                        >
+                                            {copied ? (
+                                                <IconCheck size="1rem" />
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx:150]: use <MantineIcon icon={IconCheck} /> and you can use mantine sizing there like size='md'

**Review by IrakliJani — COMMENTED** [review]
minor comments

File: packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx (hunk 2 above)
Comment (joaoviana) [reply packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx]: 🤦 we do not!

File: packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx]: No need anymore @IrakliJani , I realised I wasn't getting the data correctly, so now I've removed the state usage and reference `data` directly when displaying the expiration on successful creation of token.

File: packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx (hunk 3 above)
Comment (joaoviana) [reply packages/frontend/src/components/UserSettings/AccessTokensPanel/CreateTokenModal.tsx]: image

**Review by IrakliJani — APPROVED** [review]
looks good!

**Pull request merged**